### PR TITLE
Add CentOS 5 init scripts

### PR DESCRIPTION
**Pull request by jbergstroem**
The amount of hours spent I won't mention, but these seems to work for me.

If someone could review/test, that'd be great.

### CODE REVIEW

File: setup/centos5/resources/jenkins.initd [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+slave_start() {
+  echo Starting $NAME...
+
+  mkdir `dirname $PIDFILE` > /dev/null 2>&1 || true
+  chown $JENKINS_SLAVE_USER `dirname $PIDFILE`
+
+  ENV="PATH=$JENKINS_PATH NODE_COMMON_PIPE=$NODE_COMMON_PIPE OSTYPE=$OSTYPE"
+  runuser -m -l $JENKINS_SLAVE_USER -c "$ENV $JAVA -jar $JENKINS_SLAVE_JAR $JENKINS_SLAVE_ARGS > $JENKINS_SLAVE_LOG 2>&1 &"
+  secret=`echo $JENKINS_SLAVE_ARGS | egrep -o "\-secret \w+" | cut -d " " -f 2`
```
Comment (jbergstroem) [inline setup/centos5/resources/jenkins.initd]: Ugly. Suggestions to improve?

File: setup/centos5/README.md [hunk 2]
```diff
@@ -19,7 +19,7 @@ $ ansible-playbook -i ../ansible-inventory ansible-playbook.yaml
 Start-up of the Jenkins slaves is manual (for now), run:
 
 ```text
-$ ssh iojs-build-centos5-1 -l iojs ./start.sh
+$ ssh iojs-build-centos5-1 "service jenkins start && service jenkins-gcc41 start"
```
Comment (rvagg) [inline setup/centos5/README.md]: needs the `-l iojs` back, or `iojs@iojs-build-centos5-1`

File: setup/centos5/resources/jenkins.initd (hunk 1 above)
Comment (rvagg) [inline setup/centos5/resources/jenkins.initd]: yeah, why don't we just split it up into:
```sh
JENKINS_HOST="jenkins-iojs.nodesource.com"
JENKINS_SLAVE_NAME="iojs-digitalocean-centos5-{{id}}"
JENKINS_SECRET="{{secret}}"
```

and reconstruct `JENKINS_SLAVE_ARGS` from all of that?
+OSTYPE=linux-gnu

File: setup/centos5/resources/jenkins.initd [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+}
+
+slave_stop() {
+  echo Stopping $NAME...
+  pid=`cat $PIDFILE`
+
+  killproc -p $PIDFILE
+
+  # Wait until the monitor exits
+  while (checkpid $pid)
```
Comment (rvagg) [inline setup/centos5/resources/jenkins.initd]: would be nice to have a `-KILL` after `x` seconds, but only if you can be bothered

**Comment by rvagg** [comment]
goodo! I wouldn't mind seeing monit in here too, but that's optional

in fact, I started putting monit on some of the pi's and it basically replaced an init script because it handled all of the lifecycle itself

File: setup/centos5/resources/jenkins.initd (hunk 1 above)
Comment (jbergstroem) [inline setup/centos5/resources/jenkins.initd]: yeah, lets do that.

**Comment by jbergstroem** [comment]
Happy to use monit -- the reason I haven't really considered it is because I haven't seen jenkins die (except Windows).

File: setup/centos5/README.md (hunk 2 above)
Comment (jbergstroem) [inline setup/centos5/README.md]: `iojs` probably can't start a service. sudo make me a sandwich, I guess.

Comment (rvagg) [inline setup/centos5/README.md]: sorry, of course!

**Comment by jbergstroem** [comment]
I'm going to leave monit out of it until we run into issues. Cool?

**Comment by rvagg** [comment]
yes, cool daddy-o, :thumbsup: :watermelon:

**Comment by jbergstroem** [comment]
Ok, haven't touched anything for a while now. Considering this being stable. Will merge shortly.

**Comment by jbergstroem** [comment]
Merged in 8df1a177215ec378a56696c3717b826c17ae88e3. Thanks for the feedback making this land.

**Pull request closed without merging**